### ISSUE TITLE
Changes not detected on a manual trigger (workflow_dispatch)

### ISSUE THREAD

**Issue by TuuZzee** [dorny/paths-filter#235]
First of thanks for this action it's pretty neat!

I am using it on couple of my Github action workflow but I have one that can be trigger manually on any branch that doesn't seem to work as expected.
For some reasons the changes aren't detected, I am guessing I have configure this wrong but I am not sure where.
It's also giving me the warning: `'before' field is missing in event payload - changes will be detected from last commit
`
I have tried couple solution from other issues I found here but without success so far:
- https://github.com/dorny/paths-filter/issues/158
- https://github.com/dorny/paths-filter/issues/52

I have currently it set as such:
```yml
//...
on:
  workflow_dispatch:
//...

  checks:
//...
      - uses: dorny/paths-filter@v3
        id: filter
        with:
          base: ${{ github.ref }}
          filters: |
```

**Comment by darkxeno** [comment]
same issue here @dorny, the log show "changes will be detected from last commit" but every file is detected as a change.

**Comment by HuiNextSense** [comment]
same for me. I would expect only the last merge is compared but somehow all the commits were included. My repo is imported from another environment and the workflow is added after the import.

**Comment by melyux** [comment]
This breaks all of our workflows that can also be run manually.

**Comment by silviorp** [comment]
I was having the same problem described in the OP and this workaround worked for me!
https://github.com/dorny/paths-filter/issues/201

**Comment by saschabratton** [comment]
Yeah, I'm pretty sure it's the `base: ${{ github.ref }}` that's biting you. On a `workflow_dispatch` run `github.ref` is just your current branch (e.g. `refs/heads/main`), so you're effectively telling paths-filter to diff the branch against itself. When base == head it tries to fall back on the push event's `before` sha... but `workflow_dispatch` doesn't have a `before`, so there's nothing to compare against and it drops down to "last commit only" (that's the `'before' field is missing` warning).

Easiest fix is to just drop `base` altogether:

```yml
- uses: dorny/paths-filter@v3
  id: filter
  with:
    filters: |
      ...
```

Without `base` it defaults to your repo's default branch and diffs your branch against the merge-base with it, which is usually what you want on a manual trigger.

One caveat... if you're dispatching _on_ the default branch itself, base ends up == head again and you're back in the same spot (nothing to diff but the last commit). In that case you'd want to point `base` at a specific branch or sha that isn't the current ref.

lmk if that sorts it out 🙏🏻
